scripts/file_resume.js

```diff
@@ -0,0 +1,100 @@
+/*global $ Resumable uploadTarget uploadMessages maxSize */
+$(function() {
+
+    // This function has a server-side pair in pinc/upload_file.inc:is_valid_filename()
+    // which should be updated if the below logic changes.
+    function validate(name) {
+        // the name must contain only a-z,A-Z,0-9,-,_ and not start with -
+        // and end with .zip
+        var re = /^\w[\w-]+\.zip$/;
+        if(!re.test(name)) {
+            alert(uploadMessages.invalidChars);
+            return false;
+        }
+        if(name.length > 200) {
+            alert(uploadMessages.nameTooLong);
+            return false;
+        }
+        return true;
+    }
+
+    var resumable = new Resumable({
+        target: uploadTarget,
+        testTarget: uploadTarget,
+        forceChunkSize: true,
+        maxFiles: 1,
+        fileType: ['zip'], // use extension not mime type since zips have many
+        maxFileSize: maxSize,
+    });
+
+    function showProgress(text) {
+        $("#upload_progress").html(text);
+    }
+
+    if(resumable.support) {
+        resumable.assignBrowse($("#resumable_browse"));
+        resumable.assignDrop($("#upload_form"));
+
+        // Show the resumable div
+        $("#resumable_uploader").show();
+        // Hide the old uploader
+        $("#old_uploader").hide();
+    }
+
+    // Before we start the upload, prevent the user from hitting upload again.
+    $("#old_submit").click(function() {
+        let filename = $("#old_browse")[0].files[0].name;
+        if(validate(filename)) {
+            // won't work if we disable browse button
+            $("#old_submit").prop( "disabled", true);
+            showProgress(uploadMessages.working);
+        } else {
+            ev.preventDefault();
+        }
+    });
+
+    $("#resumable_submit").click(function() {
```

Comment:
It seems not. [(https://github.com/23/resumable.js/)] recommends using a span for the browse button but this is the submit button. I think cpeel wrote the original version of this.

Reply:
I did, but I have no memory of why I made this decision vs another one.

Reply:
It looks like they recommend span for styling reasons since they add children to the element. So if we stick with span, we should probably add an aria role as button and add keyboard handlers for enter.

Reply:
Perhaps we could create a new task for this. I've put in a note about why the span is used for the browse button.

Reply:
Sure, shouldn't hold this up I would imagine since the defect already exists.